src/mmrelay/config.py

```
@@ -574,113 +759,221 @@ def apply_env_config_overrides(config: dict[str, Any] | None) -> dict[str, Any]:
+                seen.add(debug_dir)
+                if not os.path.exists(debug_dir):
+                    continue
                 try:
-                    files = os.listdir(config_dir)
-                    logger.debug(f"Directory contents of {config_dir}: {files}")
+                    files = os.listdir(debug_dir)
+                    logger.debug("Directory contents of %s: %s", debug_dir, files)
                 except OSError:
                     pass
-            return None
+        return None
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)